### PR TITLE
Updating Hbase version to latest( 1x: 1.4.10, 2x: 2.2.0)

### PR DESCRIPTION
**Pull request by rahulKQL**
Latest version of Hbase-2.2.0([release note](https://apache.org/dist/hbase/2.2.0/RELEASENOTES.md)) has introduced new operations.

Some of the additive methods introduced in hbase 2.x:
```
BigtableAdmin#createTableAsync(TableDescriptor tableDescriptor)
BigtableAsyncAdmin#listTableDescriptors(List<TableName> list)
BigtableAsyncAdmin#restoreSnapshot(String s, boolean b, boolean b1)
BigtableAsyncAdmin#cloneSnapshot(String s, TableName tableName, boolean b)

BigtableAsyncTableRegionLocator#getRegionLocations(byte[] bytes, boolean b)
BigtableAsyncTableRegionLocator#getAllRegionLocations()
BigtableAsyncTableRegionLocator#clearRegionLocationCache()

BigtableConnection#clearRegionLocationCache()

BigtableAsyncConnection#isClosed()
BigtableAsyncConnection#setRetryPauseForCQTBE(long l, TimeUnit timeUnit)
BigtableAsyncConnection#setMaxKeyValueSize(int i)
BigtableAsyncConnection#clearRegionLocationCache()
```

After discussion, we came to a conclusion that we should implement these three methods(Other methods can be implemented based on usage and user request).

```
BigtableAdmin#createTableAsync(TableDescriptor tableDescriptor)
BigtableAsyncAdmin#listTableDescriptors(List<TableName> list)
BigtableAsyncConnection#isClosed()
```

### CODE REVIEW

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java [hunk 1]
```diff
@@ -259,6 +263,27 @@ public BigtableAsyncAdmin(CommonConnection asyncConnection) throws IOException {
     return listTables(Optional.of(pattern));
   }
 
+  @Override
+  public CompletableFuture<List<TableDescriptor>> listTableDescriptors(List<TableName> list) {
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: Please add an integration test for testing null & empty list cases and run them using the minicluster profile and bigtable profile to make sure that the beahvior matches

Comment (igorbernstein2) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: Also check if the returned list should contain column families

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java [hunk 2]
```diff
@@ -264,23 +266,18 @@ public BigtableAsyncAdmin(CommonConnection asyncConnection) throws IOException {
...
+                tableNames.stream()
                     .filter(in -> t.contains(in.getNameAsString()))
-                    .map(
-                        m ->
-                            com.google.bigtable.admin.v2.Table.newBuilder()
-                                .setName(bigtableInstanceName.toTableNameStr(m.getNameAsString()))
-                                .build())
-                    .map(Table::fromProto)
-                    .map(tableAdapter2x::adapt)
+                    .map(tbName -> getDescriptor(tbName).join())
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: There is a race condition here: the table might be deleted between a list and get operations Make sure to catch the not found error and remove the table from the list

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/async/TestAsyncAdmin.java [hunk 3]
```diff
@@ -136,18 +139,38 @@ public void testCreateTable_harness() throws Exception {
       // assertEquals(10, patTableDescriptors.get(0).getColumnFamilies()[0].getTimeToLive());
 
       // test listTableDescriptors by List<TableName>
+      try {
+        asyncAdmin.listTableDescriptors(null).get();
+        fail("listTableDescriptors should throw an exception");
+      } catch (Exception e) {
+        Assert.assertTrue(e instanceof NullPointerException);
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/async/TestAsyncAdmin.java]: Please be consistent in your use of static imports

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java (hunk 2 above)
Comment (rahulKQL) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: Thanks for detailed feedback, I have addressed this. Please take a look.

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/async/TestAsyncAdmin.java (hunk 3 above)
Comment (rahulKQL) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/async/TestAsyncAdmin.java]: Addressed

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java [hunk 4]
```diff
@@ -259,6 +265,34 @@ public BigtableAsyncAdmin(CommonConnection asyncConnection) throws IOException {
...
+                tableNames.stream()
+                    .filter(inputTableName -> t.contains(inputTableName.getNameAsString()))
+                    .map(
+                        tbName -> {
+                          try {
+                            return getDescriptor(tbName).join();
+                          } catch (CompletionException ex) {
+                            LOG.warn(
+                                "Table not found while fetching details for %s",
+                                ex, tbName.getNameAsString());
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: I don’t think you can assume that an error means the table is not found. I think you should examine the error and check if it’s a statusruntimeexception with a not found status or a notfoundexception from gax

Comment (rahulKQL) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: As [BigtableAsyncAdmin#getDescirptor](https://github.com/googleapis/cloud-bigtable-client/blob/master/bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java#L278-L296) is handling `Status.Code.NOT_FOUND`, than re throwing exception for case other than `Status.Code.NOT_FOUND` ok?

```java
toCompletableFuture(bigtableTableAdminClient.listTablesAsync())
        .thenApply(
            t ->
                tableNames.stream()
                    .filter(inputTableName -> t.contains(inputTableName.getNameAsString()))
                    .map(
                        tbName -> {
                          try {
                            return getDescriptor(tbName).join();
                          } catch (CompletionException ex) {
                            if(ex.getCause() instanceof  TableNotFoundException){
                              LOG.warn(
                                  "Table not found while fetching details for %s",
                                  ex, tbName.getNameAsString());
                              return null;
                            }
                            throw ex;
                          }
                        })
                    .filter(Objects::nonNull)
                    .collect(Collectors.toList()));
```

Comment (igorbernstein2) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: That's even better!

Comment (rahulKQL) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: Thanks for the quick response, I have updated the code. Please have a look.

Comment (igorbernstein2) [reply bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncAdmin.java]: I don't think you need to log anything...no reason to panic the user. But otherwise  LGTM
